Add words about Draft and Refined PRs to CONTRIBUTING.md (#361)
AlexsLemonade/refinebio-examples: CONTRIBUTING.md

scripts/download-data.sh
```

## Adding a new analysis

Our PR process for adding a new analysis involves two stages which are 2-3 (or more) PRs.
This splitting up an analysis into multiple PRs helps make the review process more manageable.
This process ensures that discussions around the big picture: conceptual decisions, what steps are included, which packages are used, are generally concluded before review moves on to the details and further polishing.
Note that all the following steps describe PRs to `staging` branch only (see more [about the branch set up](#pull-requests)).

### Draft PR: Big picture reviews  

- On your new git branch, [set up the analysis file from the template](#setting-up-the-analysis-file).
- Get the basic steps for the analysis set up and create a draft PR for a big picture review (Not all descriptions need to be 100% word-smithed, but the general steps/outline should be reflected).

> - Try to highlight things that encapsulate main concepts, ready for review using a `**REVIEW**` tag.
> Or use a `**DRAFT**` tag to indicate it hasn't really been worked on much yet.

Reviewer: ```suggestion
- Try to highlight things that encapsulate main concepts as ready for review using a `**REVIEW**` tag and/or use a `**DRAFT**` tag to indicate a section that hasn't really been worked on much yet.
```

I was a little confused by the wording/structure of these two sentences. What do you think of the above wording?

> - After the general outline of the analysis has been agreed upon through a reviewing process, incorporate the major feedback from the draft PR process.

Reviewer: What exactly do you mean here? From this sentence, I would think that I should incorporate the major feedback from the review on the actual draft PR and its contents, rather than incorporating the major feedback in subsequent PRs (which is fine but not necessary since it is the subsequent PRs that will be merged to staging).

```suggestion
- After the general outline of the analysis has been agreed upon through a reviewing process, incorporate the major feedback from the draft PR process in subsequent refined PRs.
```

Maybe adding some language like I did above would clarify this point a bit more (not sure we really need the word 'refined' here just yet).

Author: > incorporate the major feedback from the review on the actual draft PR

That's what I've been doing. I don't know(?) if it matters when it's incorporated as long as its incorporated before refined PRs.

Author: But I'm all for trying out a more specific approach. If you have a strategy you recommend, that might be good to put here.

Reviewer: > That's what I've been doing. I don't know(?) if it matters when it's incorporated as long as its incorporated before refined PRs.

I completely agree with you here. The main reason behind my original comment was to make sure the feedback from the draft PR review process is propagated to the appropriate subsequent PRs.
That being said, I am fine with keeping your original wording here and perhaps reiterating the incorporation of the feedback on the draft PR in subsequent PRs, in the `Refined PRs` section of this doc.

Author: I'll try to switch the wording a bit to make it clear.

- Keep the original draft PR open for easy reference.

### Refined PRs: Detailed reviews  

> - Break up the steps of the analysis into the manageable review chunks on their own branches for detailed review (you may want to discuss what ths chunks should be on the Draft PR).

Reviewer: ```suggestion
- Break up the steps of the analysis into manageable review chunks on their own branches for detailed review (you may want to discuss what the chunks should be on the Draft PR).
```
